crates/sui-swarm/src/memory/container.rs

```diff
@@ -44,12 +45,21 @@ impl Container {
         let (cancel_sender, cancel_receiver) = tokio::sync::oneshot::channel();
 
         let thread = thread::spawn(move || {
-            let span = tracing::span!(
-                tracing::Level::INFO,
-                "node",
-                name =% AuthorityPublicKeyBytes::from(config.protocol_key_pair().public()).concise(),
-            );
-            let _guard = span.enter();
+            let span = if get_global_telemetry_config()
+                .map(|c| c.enable_otlp_tracing)
+                .unwrap_or(false)
+            {
+                // we cannot have long-lived root spans when exporting trace data to otlp
```

Comment:
Whats the rationale behind this? It would be sad to loose this information

Reply:
These spans only exist to provide the per-node context in the logging output. But when exporting this as OTLP it really screws up the viewers. Spans don't get exported until they end, and these spans never end, so every other span is missing its root.

This doesn't really matter since this is only an issue when tracing `sui` or `sui-test-validator`, so I could revert this change. It was just convenient to be able to use `sui-test-validator` for testing this.